packages/typespec-azure-resource-manager/src/resource.ts

```diff
@@ -652,7 +662,15 @@ export function getResourcePathElements(
       break;
     }
 
-    if (i + 1 < segments.length && isVariableSegment(segments[i + 1])) {
+    // if the next segment is the last segment
+    if (
```

Comment:
I think if we change this logic, we will fix the issues below - it makes sense that the singleton resource name should just *always* be checked when looking for a variable segment.  

I think there is still a case to think about, when multiple resources in the resource type are singletons